### PR TITLE
Build Improvements

### PR DESCRIPTION
**Pull request by GuestSneezeOSDev**
Resolves issues, with additional improvements included such as

* https://github.com/LukeShortCloud/winesapOS/issues/916 Installed Vi
* https://github.com/LukeShortCloud/winesapOS/issues/887 Allowed using Docker or Podman for builds
* https://github.com/LukeShortCloud/winesapOS/issues/871 Read-only image supported
* https://github.com/LukeShortCloud/winesapOS/issues/866 Nvidia GFN streaming supported
* https://github.com/LukeShortCloud/winesapOS/issues/865 Xbox Game Pass has been supported
* https://github.com/LukeShortCloud/winesapOS/issues/892 Added i3 And Sway Support
* https://github.com/LukeShortCloud/winesapOS/issues/889 Added Gnome Boxes Support 
* https://github.com/LukeShortCloud/winesapOS/issues/906 Added `plasma-mobile` enviorment
* https://github.com/LukeShortCloud/winesapOS/issues/915 Added BIND support
* https://github.com/LukeShortCloud/winesapOS/issues/912 Updated Glorious eggroll version of proton

### CODE REVIEW

**Comment by GuestSneezeOSDev** [comment]
Docker is in the [`winesapos-build-repo.sh`] (https://github.com/GuestSneezeOSDev/winesapOS/blob/main/scripts/repo/winesapos-build-repo.sh#L83) I Have added Docker Desktop

I will now add `i3` support 
``

**Comment by GuestSneezeOSDev** [comment]
I will soon add https://github.com/LukeShortCloud/winesapOS/issues/906 KDE mobile support

**Comment by GuestSneezeOSDev** [comment]
I have added Sway Support

**Comment by GuestSneezeOSDev** [comment]
Add GNOME Boxes Support ([See Line 367 ](https://github.com/GuestSneezeOSDev/winesapOS/blob/main/scripts/winesapos-install.sh#L367))

**Comment by LukeShortCloud** [comment]
Thanks for submitting all of these changes! I'll be able to review them after work today.

**Comment by GuestSneezeOSDev** [comment]
I am now adding KDE Mobile

**Comment by GuestSneezeOSDev** [comment]
Add to CONTRIBUTING.md the new DE's that were added

**Comment by GuestSneezeOSDev** [comment]
KDE Mobile has been added

File: files/xcloud.desktop
Comment (LukeShortCloud) [inline files/xcloud.desktop:5]: Ditto. We need to run this with the Google Chrome Flatpak:

```
Exec=/usr/bin/flatpak run com.google.Chrome --app=https://www.xbox.com/en-US/play
```

File: scripts/repo/winesapos-build-repo.sh
Comment (LukeShortCloud) [inline scripts/repo/winesapos-build-repo.sh:82]: This repository build script is only for packages that are (1) from the AUR and (2) do not already have a Chaotic AUR package. We help fill in the gap by providing them in our winesapOS repository.

`podman-desktop` is in the Chaotic AUR so it is not needed here but `docker-desktop` is missing.

Besides that, I should have better communicated this feature request. I'm sorry for that! We already install both `docker` and `podman` (the CLI tools). I was simply looking to update the `CONTRIBUTING.md` documentation with steps on how to use `docker` and `podman` for building winesapOS.

We could still potentionally ship these `*-destop` GUI tools. The install size of `podman-desktop` is almost 350 MiB which is a bit large. I could offer it as an option to install during the first-time setup, at least. That way it doesn't take up precious space on the pre-built image.

File: scripts/winesapos-install.sh
Comment (LukeShortCloud) [inline scripts/winesapos-install.sh:575]: Nice, thank you!

`vi` also needs to be added/installed about 20 lines up right after the `echo "Installing additional packages..."` line. Basically, the top section of the `if` block is for the performance and secure images and this line here is for the minimal image.

File: scripts/winesapos-install.sh
Comment (LukeShortCloud) [inline scripts/winesapos-install.sh:794]: You can install Xorg dependencies here instead:

https://github.com/LukeShortCloud/winesapOS/blob/e97e3c5072ff19c4e89d4db59e73500e30414cbe/scripts/winesapos-install.sh#L734

File: scripts/winesapos-install.sh
Comment (LukeShortCloud) [inline scripts/winesapos-install.sh:795]: You can remove this line. The text editors are installed elsewhere.

File: scripts/winesapos-install.sh
Comment (LukeShortCloud) [inline scripts/winesapos-install.sh:800]: The actual `sway` package is missing here.

You also don't need these Xorg dependencies (Sway only works with Wayland and we install Wayland dependencies earlier on).

File: scripts/winesapos-install.sh
Comment (LukeShortCloud) [inline scripts/winesapos-install.sh:801]: Ditto. You can remove this line. The text editors are installed elsewhere.

File: scripts/winesapos-install.sh
Comment (LukeShortCloud) [inline scripts/winesapos-install.sh:814]: You're right, any package from the AUR (even if it is packaged in the Chaotic AUR or winesapOS repository) should use `yay`. It's our fail-safe in case all repositories disappeared for whatever reason.

This is kind of implied so you don't need the comment here. I should probably make a note of it in the `CONTRIBUTING.md` guide.

File: scripts/winesapos-install.sh
Comment (LukeShortCloud) [inline scripts/winesapos-install.sh:815]: I was surprised to see that was not packaged in the Chaotic AUR repository. It turns out that they are waiting for the `plasma-dialer` package to be fixed upstream.

`https://github.com/chaotic-aur/packages/issues/3052`

That's okay, we can still add this to winesapOS. The only change is that you don't need `cmake` here. We install it much earlier on in the install process since a lot of AUR packages need it.

File: scripts/winesapos-read-only.sh
Comment (LukeShortCloud) [inline scripts/winesapos-read-only.sh:2]: Simple and effective! I still want to go with a different solution long-term that will help with day 2 operations of upgrades and keeping consistency. For now, users could potentially enable this read-only service if they want it, disable it to run a winesapOS Upgrade, and then re-enable it again.

I don't think the first-time setup and other applications will properly work in this mode but we can add it for now and users/developers can test it out.

**Review by LukeShortCloud — CHANGES_REQUESTED** [review]
Hey @GuestSneezeOSDev , thanks for all of these awesome additions! I tested most everything out (Xbox Cloud Gaming, GeForce Now, i3, and Sway) and it seems to be working. Hopefully my feedback is not too overwhelming. All of the changes needed are minor.

**Comment by LukeShortCloud** [comment]
Once we get this merged in, I'll probably hold off on new features (besides having recommended defaults for the first-time setup which I'm still working on). That way we can get about a week of testing in before the final release on August 18th. After that date, we can start development of winesapOS 4.2.0 and I can help out more with Super-QuantamOS. :-)

File: files/gfn.desktop
Comment (LukeShortCloud) [reply files/gfn.desktop:3]: The Geforce also needs a capitalized F: GeForce.

File: README.md
Comment (GuestSneezeOSDev) [reply README.md:177]: Thanks

File: files/gfn.desktop
Comment (GuestSneezeOSDev) [reply files/gfn.desktop]: Thanks

File: files/gfn.desktop
Comment (GuestSneezeOSDev) [reply files/gfn.desktop]: Thanks mate

File: files/gfn.desktop
Comment (GuestSneezeOSDev) [reply files/gfn.desktop:4]: Thanks

File: scripts/repo/winesapos-build-repo.sh
Comment (GuestSneezeOSDev) [reply scripts/repo/winesapos-build-repo.sh:82]: I Will remove the GUI `*-desktop` packages and use the `CLI` packages

File: scripts/winesapos-install.sh
Comment (GuestSneezeOSDev) [reply scripts/winesapos-install.sh:794]: Thanks

File: scripts/winesapos-read-only.sh
Comment (GuestSneezeOSDev) [reply scripts/winesapos-read-only.sh:2]: maybe make a desktop shortcut which activates / disables the `read-only` service

Comment (GuestSneezeOSDev) [reply scripts/winesapos-read-only.sh:2]: https://github.com/GuestSneezeOSDev/winesapOS/blob/main/files/winesap-read-only.service

**Comment by GuestSneezeOSDev** [comment]
I will now add bind

File: scripts/winesapos-install.sh
Comment (LukeShortCloud) [inline scripts/winesapos-install.sh:793]: nit: there is an extra whitespace behind `echo`. It should align with the code below.

File: scripts/winesapos-install.sh
Comment (LukeShortCloud) [inline scripts/winesapos-install.sh:794]: nit: There is an extra space between these two package names: `i3-wm  i3lock`.

File: scripts/winesapos-read-only.sh
Comment (LukeShortCloud) [reply scripts/winesapos-read-only.sh:2]: Good idea. I'll try that out.

**Review by LukeShortCloud — CHANGES_REQUESTED** [review]
Thanks for your changes and the addition of `bind`! We're almost there. :-)

File: scripts/repo/winesapos-build-repo.sh
Comment (GuestSneezeOSDev) [reply scripts/repo/winesapos-build-repo.sh:81]: Thanks

**Comment by GuestSneezeOSDev** [comment]
if you need to make it `rw` again just replace `ro` with `rw`

**Comment by GuestSneezeOSDev** [comment]
Updated GE version https://github.com/LukeShortCloud/winesapOS/issues/912

**Comment by GuestSneezeOSDev** [comment]
I would like to know how could I use the `bazzite` kernel

**Comment by LukeShortCloud** [comment]
The `linux-fsync-nobara-bin` package on the AUR provides the Linux kernel used by Bazzite and Nobara. It also includes the `*-headers` files so DKMS modules (ex., NVIDIA) can be built against it.

The `fsync` name is misleading as that is just one of many patches they provide. It is the single best improvement to winesapOS in recent times! Hopefully it can help you, too.

https://pagure.io/kernel-fsync/blob/main/f/SOURCES

I would have linked directly to the related AUR page but it is currently down right now.

https://downforeveryoneorjustme.com/aur.archlinux.org

**Comment by LukeShortCloud** [comment]
I tested the read-only functionality and, in practice, it doesn't work. The first problem is that it needs to be run with root privileges but that's an easy fix. The second problem is that `mount` won't let me change the root partition to read-only. My best guess is because a lot of services are writing to the disk in the background (which is normal for an operating system).

```
$ sudo mount -o remount,ro /
mount: /: mount point is busy
       dmesg(1) may have more information after failed mount system call.
```

`dmesg` does not have anything useful to report.

I would use something like `rpm-ostree` for read-only support but that is only for RPM-based Linux distributions (not Arch Linux). [bootc](https://github.com/containers/bootc) is the spiritual successor and supports any operating system. That is one potential avenue. Read-only support is a long-term goal for winesapOS 5.0.0. We'll need to keep doing some research and testing in the meantime.

**Comment by GuestSneezeOSDev** [comment]
Perhaps you can try
```
sudo mount -o remount,ro,force /
```

**Comment by GuestSneezeOSDev** [comment]
I would also like to ask if i could implement a new image type `ChimeraOS` or [`EndeavourOS`](https://github.com/endeavouros-team)

**Comment by LukeShortCloud** [comment]
Using the `force` mount option doesn't work with Btrfs:

```
$ sudo mount -o remount,ro,force /
mount: /: fsconfig system call failed: btrfs: Unknown parameter 'force'.
       dmesg(1) may have more information after failed mount system call.
```

Passing `-f` force flag to mount seems to work at first but it actually silently fails.

```
$ sudo mount -o remount,ro / -f
$ mount | grep btrfs | grep ro
```

I'm good friends with the ChimeraOS folks. One of the read-only solutions we're looking into is their [frzr](https://github.com/ChimeraOS/frzr) project. It's going through a big rewrite right now. bootc is in alpha. It'll take some time for these to mature so I think we should simply wait.

**Comment by LukeShortCloud** [comment]
Somewhat related to this, ChimeraOS might be helping us develop [archupgrade-go](https://github.com/ohaiibuzzle/archupgrade-go). The primary goal is to automate upgrades. The secondary goal is to have target states. For example, there could be a target for "gaming desktop" (basically winesapOS) and a target for "gaming handheld" (basically ChimeraOS).

**Comment by GuestSneezeOSDev** [comment]
I tested this works with Super QuantamOS and Arch Linux
```
echo "GRUB_CMDLINE_LINUX_DEFAULT="loglevel=3 quiet ro"" >> /boot/grub/grub.cfg
sudo reboot
```

**Comment by LukeShortCloud** [comment]
That configuration normally goes in `/etc/default/grub`. Then you need to run `grub-mkconfig -o /boot/grub/grub.cfg`.

I did a quick test and make sure to replace the `rw` boot argument with `ro`. I get a somewhat scary message saying that the file system is not meant to be read-write and that I might need to do a FSCK.

Once booted, it's still `rw` anyways. I'm not sure why this is the behavior in winesapOS.

**Comment by GuestSneezeOSDev** [comment]
which session of winesapOS are you using (by session I mean which version and are you running it off a usb or external HDD / SSD)

**Comment by LukeShortCloud** [comment]
winesapOS 4.1.0-beta.2 as a virtual machine (QEMU/KVM via Virtual Machine Manager). It's a virtual SSD using the VirtIO SCSI driver.

**Comment by GuestSneezeOSDev** [comment]
if your using it on a USB it is most likely because winesapOS is a portable OS

**Comment by GuestSneezeOSDev** [comment]
Perhaps we should use [`frzr`](https://github.com/ChimeraOS/frzr) instead

**Comment by LukeShortCloud** [comment]
Yeah, we might end up using frzr.

Read-only support is not a priority right now. I owe the community a winesapOS 4.1.0 release soon and it has already been 6 months in the making (we usually do releases every 3 months).

We can continue to plan for winesapOS 5.0.0 / the read-only update here: https://github.com/LukeShortCloud/winesapOS/issues/871

**Comment by GuestSneezeOSDev** [comment]
I will comment the `.sh` file and delete the service file

**Comment by LukeShortCloud** [comment]
We should probably delete the `scripts/winesapos-read-only.sh` since (1) it's small, (2) doesn't work, and (3) won't be used yet. This PR has a history associated with the changes and our conversation so we can always refer back to this page.

Unless you have anything else you wanted to add, this all looks good to merge today! :-)

**Review by LukeShortCloud — APPROVED** [review]

**Comment by LukeShortCloud** [comment]
It's all been merged, thank you! I would not have had time to do all that myself for this upcoming release. I appreciate your time!

**Comment by GuestSneezeOSDev** [comment]
Welcome mate

**Pull request merged**
